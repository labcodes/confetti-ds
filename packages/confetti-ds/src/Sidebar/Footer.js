import React from "react";
import PropTypes from "prop-types";

export default class Footer extends React.Component {
  static propTypes = {
    /** Components that will be rendered in the Footer. */
    children: PropTypes.node,
  };

  static defaultProps = {
    children: undefined,
  };

  render() {
    const { children } = this.props;
    return <div className="lab-narrow-sidebar__footer">{children}</div>;
  }
}
