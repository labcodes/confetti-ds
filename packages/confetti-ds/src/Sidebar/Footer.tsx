import React from "react";
import PropTypes from "prop-types";

/**
 *
 * @param children Components that will be rendered in the Footer.
 * @returns {JSX.Element}
 * @constructor
 */
export default function Footer({ children }) {
  return <div className="lab-narrow-sidebar__footer">{children}</div>;
}

Footer.propTypes = {
  /** Components that will be rendered in the Footer. */
  children: PropTypes.node,
};

Footer.defaultProps = {
  children: undefined,
};
