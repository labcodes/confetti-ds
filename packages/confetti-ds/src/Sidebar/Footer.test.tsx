import React from "react";
import renderer from "react-test-renderer";

import Footer from "./Footer";

describe("Footer", () => {
  it("renders with base props", async () => {
    const renderedComponent = renderer
      .create(<Footer>Test content</Footer>)
      .toJSON();
    expect(renderedComponent).toMatchSnapshot();
  });
});
