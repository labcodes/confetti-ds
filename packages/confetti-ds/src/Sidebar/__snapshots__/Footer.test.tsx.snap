// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer renders with base props 1`] = `
<div
  className="lab-narrow-sidebar__footer"
>
  Test content
</div>
`;
